jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py

```diff
@@ -654,32 +705,290 @@ def _set_value_in_subset_definition(self, index, name, desired_key, new_value):
+        region : region, list of region objects, or str
+            A region object can be one of the following:
+
+            * Astropy ``regions`` object
+            * ``photutils`` apertures (limited support until ``photutils``
+              fully supports ``regions``)
+            * specutils ``SpectralRegion`` object
+
+            A string which represents a ``regions`` or ``SpectralRegion`` file.
+            If given as a list, it can only contain spectral or non-spectral regions, not both.
+
+        combination_mode : list, str, or `None`
+            The way that regions are created or combined. If a list, then it must be the
+            same length as regions. If `None`, then it will follow the default glue
+            functionality for subset creation.
+
+        max_num_regions : int or `None`
+            Maximum number of regions to load, starting from top of the list.
+            Default is to load everything.  If you are providing a large file/list
+            input for ``region``, it is recommended
+
+        refdata_label : str or `None`
+            **This is only applicable to non-spectral regions.**
+            Label of data to use for sky-to-pixel conversion for a region, or
+            mask creation. Data must already be loaded into Jdaviz.
+            If `None`, defaults to the reference data in the default viewer.
+            Choice of this data is particularly important when sky
+            region is involved.
+
+        return_bad_regions : bool
+            If `True`, return the regions that failed to load (see ``bad_regions``);
+            This is useful for debugging. If `False`, do not return anything (`None`).
+
+        Returns
+        -------
+        bad_regions : list of (obj, str) or `None`
+            If requested (see ``return_bad_regions`` option), return a
+            list of ``(region, reason)`` tuples for region objects that failed to load.
+            If all the regions loaded successfully, this list will be empty.
+            If not requested, return `None`.
+
         """
-        viewer_name = viewer or self.app._jdaviz_helper._default_spectrum_viewer_reference_name
-        range_viewer = self.app.get_viewer(viewer_name)
-        for index, sub_region in enumerate(spec_region):
-            if isinstance(mode, list):
-                if len(mode) != (len(spec_region) - 1):
-                    raise ValueError("list of mode must be size of spec_region minus one")
-                if index == 0:
-                    m = NewMode
-                else:
-                    m = mode[index - 1]
+        if isinstance(region, str):
+            if os.path.exists(region):
+                from regions import Regions
+                region_format = kwargs.pop('region_format', None)
+                try:
+                    raw_regs = Regions.read(region, format=region_format)
+                except Exception:  # nosec
+                    raw_regs = SpectralRegion.read(region)
```

Comment:
This is fine for now, but I wonder if there's any way we could predict in advance instead of try/except.  As it is now, checking the extension would probably work, but in the future we may have the same extension supported between spectral + spatial, so that isn't a longterm solution.